WordPress/Classes/Services/BlogService.m

```diff
@@ -220,23 +220,37 @@ - (void)syncSettingsForBlog:(Blog *)blog
                     success:(void (^)())success
                     failure:(void (^)(NSError *error))failure
 {
-    id<BlogServiceRemote> remote = [self remoteForBlog:blog];
-    [remote syncSettingsForBlog:blog
-                        success:^(RemoteBlogSettings *settings) {
-                            blog.blogName = settings.name;
-                            blog.blogTagline = settings.desc;
-                            if (settings.defaultCategory) {
-                                blog.defaultCategoryID = settings.defaultCategory;
-                            }
-                            if (settings.defaultPostFormat) {
-                                blog.defaultPostFormat = settings.defaultPostFormat;
-                            }
-                            [self.managedObjectContext save:nil];
-                            if (success) {
-                                success();
-                            }
-                        }
-                        failure:failure];
+    NSManagedObjectID *blogID = [blog objectID];
+    [self.managedObjectContext performBlock:^{
```

Comment:
The way services should be used is that the place that creates the service instance is responsible for wrapping the method call in the performBlock. So I would go back to the original implementation and make sure all the places that call this code (that aren't obvious to be on the main thread) wrap it in the performBlock there.  @jleandroperez do you agree?